packages/renderers-rust/README.md

```diff
@@ -37,12 +37,114 @@ codama.accept(renderVisitor(pathToGeneratedFolder, options));
+| `dependencyMap`               | `Record<string, string>`                                                                                                | `{}`                    | A mapping between import aliases and their actual crate name or path in Rust.                                                                                                                    |
+| `renderParentInstructions`    | `boolean`                                                                                                               | `false`                 | When using nested instructions, whether the parent instructions should also be rendered. When set to `false` (default), only the instruction leaves are being rendered.                          |
+| `traitOptions`                | [`TraitOptions`](#trait-options)                                                                                        | `DEFAULT_TRAIT_OPTIONS` | A set of options that can be used to configure how traits are rendered for every Rust types. See [documentation below](#trait-options) for more information.                                     |
+
+## Trait Options
+
+The Rust renderer provides sensible default traits when generating the various Rust types you client will use. However, you may wish to configure these traits to better suit your needs. The `traitOptions` attribute is here to help you with that. Let's see the various settings it provides.
+
+### Default traits
+
+Using the `traitOptions` attribute, you may configure the default traits that will be applied to every Rust type. These default traits can be configured using 4 different attributes:
+
+-   `baseDefaults`: The default traits to implement for all types.
+-   `dataEnumDefaults`: The default traits to implement for all data enum types, in addition to the `baseDefaults` traits. Data enums are enums with at least one non-unit variant — e.g. `pub enum Command { Write(String), Quit }`.
+-   `scalarEnumDefaults`: The default traits to implement for all scalar enum types, in addition to the `baseDefaults` traits. Scalar enums are enums with unit variants only — e.g. `pub enum Feedback { Good, Bad }`.
+-   `structDefaults`: The default traits to implement for all struct types, in addition to the `baseDefaults` traits.
+
+Note that you must provide the fully qualified name of the traits you provide (e.g. `serde::Serialize`). Here are the default values for these attributes:
+
+```ts
+const traitOptions = {
+    baseDefaults: [
+        'borsh::BorshSerialize',
+        'borsh::BorshDeserialize',
+        'serde::Serialize',
+        'serde::Deserialize',
+        'Clone',
+        'Debug',
+        'Eq',
+        'PartialEq',
+    ],
+    dataEnumDefaults: [],
+    scalarEnumDefaults: ['Copy', 'PartialOrd', 'Hash', 'num_derive::FromPrimitive'],
+    structDefaults: [],
+};
```

Comment:
I think there's a case to be made for the default being zero traits and zero scalar enum configurations, but I don't feel too strongly about it.

Reply:
We had to keep them like that in order to avoid introducing breaking changes. This produces code in the exact same way we were before. But I think these are sensible defaults for most clients, right?

Reply:
Sure, it doesn't really hurt to just let people change the base defaults anyway, and avoid the breaking change.